### ISSUE TITLE
[FEATURE REQUEST]Add order Hints for Bulk Copy operations

### ISSUE THREAD

**Issue by luxu1-ms** [microsoft/mssql-jdbc#1481]
## Is your feature request related to a problem? If so, please give a short summary of the problem and how the feature would resolve it

Can you add support for order hints for bulk copy operations? I am going to add order hints for [spark-sql-connector ](https://github.com/microsoft/sql-spark-connector)but there is no implementation of order hints for SQLServerBulkCopy.class, SQLServerBulkCopyOptions.class
## Reference Documentations/Specifications

[BULK INSERT](https://docs.microsoft.com/en-us/sql/t-sql/statements/bulk-insert-transact-sql?view=sql-server-ver15) allow to specify if data already has a specific order so that Azure SQL / SQL Server can avoid sorting it again if inserting data into a clustered index.

ORDER ( { column [ ASC | DESC ] } [ ,... n ] ) Specifies how the data in the data file is sorted. Bulk import performance is improved if the data being imported is sorted according to the clustered index on the table, if any. If the data file is sorted in a different order, that is other than the order of a clustered index key or if there is no clustered index on the table, the ORDER clause is ignored. The column names supplied must be valid column names in the destination table. By default, the bulk insert operation assumes the data file is unordered. For optimized bulk import, SQL Server also validates that the imported data is sorted.

https://docs.microsoft.com/en-us/sql/connect/ado-net/sql/bulk-copy-order-hints?view=sql-server-ver15

**Comment by ulvii** [comment]
Hi @luxu1-ms ,
Thank you for the suggestion. The team will take a look into this.

**Comment by luxu1-ms** [comment]
Hi, any update on this issue

**Comment by lilgreenbird** [comment]
hi @luxu1-ms 
This issue is in our backlog. This will be triaged when we plan for the next semester.

Thanks.

**Comment by shivsood** [comment]
Any updates on this?

**Comment by lilgreenbird** [comment]
hi @shivsood 

This issue is still in our backlog. It didn't make the cut as we have to prioritize bug fixes and other features that are in higher demand. Please feel free to fill out our [survey ](https://forms.office.com/Pages/ResponsePage.aspx?id=DQSIkWdsW0yxEjajBLZtrQAAAAAAAAAAAAFhcKCP4bxUQUxBNDA1NjRER0Y5SVhRQ1hSN0g1Q09MQS4u) for features you would like to see in the driver we do look at the results when we do semester planning.

**Comment by siggemannen** [comment]
Would patches for this be accepted? it looks to be literally a couple of rows of changes since it's all in the query being send to sql server + new orderoption parameter to bulk copy options

**Comment by Jeffery-Wasty** [comment]
Yes, we accept external PRs. Feel free to submit something, and we'll take a look.

**Comment by atul-delphix** [comment]
Hi @lilgreenbird ,

Any update on this PR to merge & release ?
